nautobot/ipam/signals.py

```diff
@@ -65,12 +65,23 @@ def ip_address_to_interface_pre_delete(instance, raw=False, **kwargs):
     # that is the primary_v{version} of the host machine.
 
     if getattr(instance, "interface"):
-        host = instance.interface.device
-        other_assignments_exist = (
-            IPAddressToInterface.objects.filter(interface__device=host, ip_address=instance.ip_address)
-            .exclude(id=instance.id)
-            .exists()
-        )
+        interface = instance.interface
+        host = interface.device or interface.module.device
+
+        if interface.device:
+            # Check assignments within the device
+            other_assignments_exist = (
+                IPAddressToInterface.objects.filter(interface__device=host, ip_address=instance.ip_address)
+                .exclude(id=instance.id)
+                .exists()
+            )
+        elif interface.module:
+            # Check assignments within the module
+            other_assignments_exist = (
+                IPAddressToInterface.objects.filter(interface__module=interface.module, ip_address=instance.ip_address)
+                .exclude(id=instance.id)
+                .exists()
+            )
```

Comment:
I'd think we'd want this to actually be inclusive (check *both* possibilities, as for example I think it should be OK to remove an IP from an interface directly on the Device if that IP is also assigned to an interface on a Module on that same Device. Also need to consider other modules (potentially nested) on the same device. I think this should maybe be something like:

```suggestion
        other_assignments_exist = (
            IPAddressToInterface.objects.filter(interface__in=host.all_interfaces(), ip_address=instance.ip_address)
            .exclude(id=instance.id)
            .exists()
        )
```

Reply:
seems to work after testing with nested modules and module on device.